apis/v1beta1/providerconfig_types.go

```diff
@@ -21,6 +21,23 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	// BasicAuth gitlab BasicAuth method of authentification
+	BasicAuth AuthType = "BasicAuth"
+
+	// JobToken gitlab JobToken method of authentification
+	JobToken AuthType = "JobToken"
+
+	// OAuthToken gitlab OAuthToken method of authentification
+	OAuthToken AuthType = "OAuthToken"
+
+	// PrivateToken gitlab PrivateToken method of authentification.
+	PrivateToken AuthType = "PrivateToken"
```

Comment:
I think the general term is personal access token
```suggestion
	PersonalAccessToken AuthType = "PersonalAccessToken"
```